lean/main/overview.lean

```diff
@@ -0,0 +1,200 @@
+
+Throughout this book you will see references to the elaborator; and in the "Extra: Pretty Printing" chapter you can read on delaborators.
+
+## 3 essential functions and their syntax sugars
+
+Now, when you're reading Lean source code, you will see 11(+?) commands specifying the **parsing**/**elaboration**/**evaluation** process.  Most of them are syntax sugar, and you only need **3 commands** to do metaprogramming in Lean:
+
+- **syntax rule: `syntax (name := xxx) ... : command`**
+
+    For example, `syntax (name := xxx) "#help" "option" (ident)? : command` is a syntax rule that will match (remember the regex analogy) all strings in the form of `"#help option hi.hello"` or just `"#help option"`.   
+    Other widespread syntax categories are `tactic` and `term`, all of these are used in different physical places in your code.
+
+- **macro: `@[macro xxx] def ourMacro : Macro := ...`**
+
+    For example, ``@[macro xxx] def ourMacro : Macro := λ stx => match stx with | _ => `(tactic| pick_goal 2)``. Whenever your Lean code matches the syntax rule with the name `xxx`, this macro will be applied.  
+    If the syntax rule was `syntax (name := xxx) "swap" : tactic`, this macro will syntactically turn `swap` into `pick_goal 2` - and the subsequent elaboration step will be handled by the `pick_goal 2` syntax rule.
+
+- **elab: `@[command_elab xxx] def ourElab : CommandElab := ...`**
+
+    For example, ``@[term_elab our_help] def ourElab : TermElab := λ stx tp => (Expr.app (Expr.const `Nat.add []))``.  
+    Our `elab` function can be of different types - the **CommandElab** and **TermElab** that you saw already, and **Tactic**.  
+    **TermElab** stands for **Syntax → Option Expr → TermElabM Expr**, so the elaboration function is expected to return the **Expr** object.  
+    **CommandElab** stands for **Syntax → CommandElabM Unit**, so it shouldn't return anything.  
+    **Tactic** stands for **Syntax → TacticM Unit**, so it shouldn't return anything either.  
+    This corresponds to out intuitive understanding of terms, commands and tactics in Lean - terms return a particular value upon execution, commands modify the environment or print something out, and tactics modify the proof state.
+
+These `syntax (name := xxx) ... : command`, `@[macro xxx] def ourMacro : Macro := ...` and `@[command_elab xxx] def ourElab : CommandElab := ...` are the 3 essential, low-level commands, and you can get away with them only. Lean standard library and Mathlib use many syntax sugars, however, so memorizing them is well worth the effort. I'm summing them up in the next diagram.
```

Comment:
`Lean standard library and Mathlib use many syntax sugars, however, so memorizing them is well worth the effort` not sure what you meant here